### PR TITLE
[addons] prevent unneccesary dependency install dialog popup

### PR DESCRIPTION
**Pull request by howie-f**
## Description
this prevents popup of `"The following additional add-ons will be installed"` when only modules are missing, as they are usually not visible to the user.

second commit uses `std::move` instead of copying vectors around

## Motivation and Context
while installing `Tubed` i encountered a dependency install dialog announcing installation of `inputstream.adaptive` which was already installed.

## How Has This Been Tested?
debian buster

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [X] **Improvement** (non-breaking change which improves existing functionality)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document

@phunkyfish mind having a look at it?

### CODE REVIEW

**Comment by phunkyfish** [comment]
How do modules relate to inputstream.adaptive which is an add-on?

**Comment by howie-f** [comment]
> How do modules relate to inputstream.adaptive which is an add-on?

assume `inputstream.adaptive` is "built in manually" like it is on android for example, or user installed it beforehand.
now you install `Tubed` from the repo which has dependencies:
```
inputstream.adaptive
script.module.x   -> not installed yet
script.module.y   -> also not installed
```

in that case you'll see `following addons will be installed: inputstream.adaptive`...
not true, that is already installed

**Comment by phunkyfish** [comment]
So if any modules are required to be installed only non module dependencies will be shown in the dialog.

So the solution is to not show the dialog at all in the case. Gotcha

**Comment by howie-f** [comment]
exactly .. right

**Comment by howie-f** [comment]
jenkins build this please

**Comment by howie-f** [comment]
jenkins build this please

**Comment by howie-f** [comment]
jenkins build this please

**Comment by howie-f** [comment]
if there are no objections i would merge this tomorrow. saves users from one or the other annoying 'false alert' install dialog.

**Comment by phunkyfish** [comment]
Please wait for an approval. I’ll review shortly.

File: xbmc/addons/gui/GUIDialogAddonInfo.cpp
Comment (phunkyfish) [inline xbmc/addons/gui/GUIDialogAddonInfo.cpp:780]: A comment would make sense here. I'm not entirely sure why we should show deps on install if an addon is not available in repos?

Comment (howie-f) [reply xbmc/addons/gui/GUIDialogAddonInfo.cpp:780]: so the user will get a chance to see why the following installation fails. as the „not available“ dialog pops up right after clicking the install button. best option would be the installation is cancelled in this case.

Comment (howie-f) [reply xbmc/addons/gui/GUIDialogAddonInfo.cpp:780]: would be possible to disable the install button if a dep is unavailable. but then nobody has a clue why it‘s disabled unless they look under dependencies...

Comment (phunkyfish) [reply xbmc/addons/gui/GUIDialogAddonInfo.cpp:780]: Definitely needs a comment in the code!!!

**Review by phunkyfish — APPROVED** [review]
LGTM

**Comment by phunkyfish** [comment]
jenkins build this please

**Pull request merged**
